### PR TITLE
Feature/ff 106

### PR DESCRIPTION
**Pull request by open-schnick**

### CODE REVIEW

**Comment by open-schnick** [comment]
Tests need to be fixed, Unit test to be written. @qvalentin @Gimleux  can you write more scenarios that you want to test e.g. user groups, when an admin user wants to change a different user etc..

File: src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java [hunk 1]
```diff
@@ -130,14 +141,93 @@ public void registerNewUser(UserRegisterForm newUser) {
...
+
+        UserEntity userEntityToUpdate = userRepository.findByUserId(userId);
+        Update newUpdate = new Update();
+        boolean changesWereMade = false;
+
+        // username
+        String username = userToUpdate.getUsername();
+        if (null != username) {
+            if (!stringIsValid(username))
+                throw new UserNotUpdatedException("Wanted to change username, but username was not valid.");
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java:173]: but new username..

File: src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java [hunk 2]
```diff
@@ -130,14 +141,93 @@ public void registerNewUser(UserRegisterForm newUser) {
...
+            changesWereMade = true;
+            newUpdate.set("groupIds", userToUpdate.getGroupIds());
+        }
+
+        if(!changesWereMade)
+            throw new UserNotUpdatedException("No changes were made.");
+
+        Query query = new Query();
+        query.addCriteria(Criteria.where("userId").is(userId));
+        mongoTemplate.findAndModify(query, newUpdate, UserEntity.class);
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java:230]: +1

File: src/test/java/de/filefighter/rest/TestUtils.java [hunk 3]
```diff
@@ -14,7 +14,7 @@ public static String serializeUserRequest(String confirmationPassword, int[] gro
             jsonString.append("\"groupIds\": ").append(Arrays.toString(groupIds)).append(",");
         }
         if (password != null) {
-            jsonString.append("\"password\": \"").append(password).append(username != null?"\",":"");
+            jsonString.append("\"password\": \"").append(password).append(username != null ? "\"," : "\"");
```
Comment (qvalentin) [inline src/test/java/de/filefighter/rest/TestUtils.java:17]: ups

**Review by qvalentin — APPROVED** [review]

**Pull request merged**